api/client/container/run.go

```diff
@@ -270,32 +268,58 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 	var status int
 
 	// Attached mode
-	if *flAutoRemove {
+	if opts.autoRemove {
 		// Autoremove: wait for the container to finish, retrieve
 		// the exit code and remove the container
-		if status, err = cli.client.ContainerWait(ctx, createResponse.ID); err != nil {
+		if status, err = client.ContainerWait(ctx, createResponse.ID); err != nil {
 			return runStartContainerErr(err)
 		}
-		if _, status, err = cli.getExitCode(ctx, createResponse.ID); err != nil {
+		if _, status, err = dockerCli.GetExitCode(ctx, createResponse.ID); err != nil {
 			return err
 		}
 	} else {
 		// No Autoremove: Simply retrieve the exit code
 		if !config.Tty {
 			// In non-TTY mode, we can't detach, so we must wait for container exit
-			if status, err = cli.client.ContainerWait(ctx, createResponse.ID); err != nil {
+			if status, err = client.ContainerWait(ctx, createResponse.ID); err != nil {
 				return err
 			}
 		} else {
 			// In TTY mode, there is a race: if the process dies too slowly, the state could
 			// be updated after the getExitCode call and result in the wrong exit code being reported
-			if _, status, err = cli.getExitCode(ctx, createResponse.ID); err != nil {
+			if _, status, err = dockerCli.GetExitCode(ctx, createResponse.ID); err != nil {
 				return err
 			}
 		}
 	}
 	if status != 0 {
-		return Cli.StatusError{StatusCode: status}
+		return cli.StatusError{StatusCode: status}
 	}
 	return nil
 }
+
+// reportError is a utility method that prints a user-friendly message
+// containing the error that occurred during parsing and a suggestion to get help
+func reportError(stderr io.Writer, name string, str string, withHelp bool) {
```

Comment:
I think this is trying to mimic the error message that is normally raised when a flag or value is incorrect. These two commands do a bunch of validation on the flag values after the initial parsing, where as I think many commands don't do any additional validation.